File: database-driver-api/src/main/java/de/lino/database/export/transcript/TranscriptLegendEntry.java
package de.lino.database.export.transcript;

/**
 * One entry of a closing legend, shared by every built-in {@link TranscriptExporter}
 * implementation - {@code ExportCoordinator.TranscriptPDFExporter},
 * {@code ExportCoordinator.TranscriptExcelExporter}, {@code ExportCoordinator.TranscriptCSVExporter},
 * {@code ExportCoordinator.TranscriptXMLExporter}, {@code ExportCoordinator.TranscriptJsonExporter}
 * and {@code ExportCoordinator.TranscriptDocxExporter} - e.g. a grading-scale key:
 * {@code label} in a left column, {@code description} in a right column, aligned
 * consistently across every entry.
 *
 * @param label the entry's left-column text, e.g. a grade range
 * @param description the entry's right-column text, e.g. what the grade range means
 */
public record TranscriptLegendEntry(String label, String description) {
}

File: python/database-driver-api/src/database_driver/api/utils/export/transcript/transcript_legend_entry.py
"""Mirror of ``de.lino.database.utils.export.transcript.TranscriptLegendEntry``."""

from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True, slots=True)
class TranscriptLegendEntry:
    """One entry of a closing legend, shared by every built-in ``TranscriptExporter``
    implementation - e.g. a grading-scale key: ``label`` in a left column,
    ``description`` in a right column, aligned consistently across every entry.

    Attributes:
        label: The entry's left-column text, e.g. a grade range.
        description: The entry's right-column text, e.g. what the grade range means.
    """

    label: str
    description: str
